### PR TITLE
*: add tip in contributing.md

### LINKED ISSUE #12142: Add the tip about `ulimit` in doc
## Development Task

I found the process maximum number is a problem especially for wsl user.

 Add the tip about `ulimit` in doc

### PR DESCRIPTION
**Pull request by jackwener**
Signed-off-by: jackwener <jakevingoo@gmail.com>

### What is changed and how it works?

Issue Number: Close #12142

What's Changed:

```commit-message
polish contributing.md
```

### Related changes

None

### Check List 

- No code

Side effects

None

### Release note 

```release-note
None
```

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -105,7 +105,13 @@ When building with make, cargo will automatically use [pipelined][p] compilation
 
 ## Running TiKV
 
-To run TiKV as an actual key-value store, you will need to run it as a cluster (a cluster can have just one node, which is useful for testing). You can do this on a single machine or on multiple machines. You need to use [PD](https://github.com/tikv/pd) to manage the cluster (even if there is just one node on a single machine). Instructions are in our [docs](https://tikv.org/docs/dev/tasks/deploy/binary/) (if you build TiKV from source, then you don't need to download the binary).
+To run TiKV as an actual key-value store, you will need to run it as a cluster (a cluster can have just one node, which is useful for testing). You can do this on a single machine or on multiple machines. 
+
+Install [PD](https://github.com/tikv/pd) to manage the cluster (even if just one node on a single machine).
```
Comment (sticnarf) [inline CONTRIBUTING.md:110]: May I ask why you change the wording to "install"? "Install" sounds like a strong instruction but we don't actually care about it.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -105,7 +105,13 @@ When building with make, cargo will automatically use [pipelined][p] compilation
...
 ## Running TiKV
 
-To run TiKV as an actual key-value store, you will need to run it as a cluster (a cluster can have just one node, which is useful for testing). You can do this on a single machine or on multiple machines. You need to use [PD](https://github.com/tikv/pd) to manage the cluster (even if there is just one node on a single machine). Instructions are in our [docs](https://tikv.org/docs/dev/tasks/deploy/binary/) (if you build TiKV from source, then you don't need to download the binary).
+To run TiKV as an actual key-value store, you will need to run it as a cluster (a cluster can have just one node, which is useful for testing). You can do this on a single machine or on multiple machines. 
+
+Install [PD](https://github.com/tikv/pd) to manage the cluster (even if just one node on a single machine). 
+
+Instructions are in our [docs](https://tikv.org/docs/dev/tasks/deploy/binary/) (if you build TiKV from source, you  could skip `1. Download package` and `tikv-server` is in directory `/target`).
+
+Tips: Run TiKV need `ulimit` >= 82920. WSL2 should notice BerserkerGaruk's solution in [issue](https://github.com/Microsoft/WSL/issues/1688)
```
Comment (sticnarf) [inline CONTRIBUTING.md:114]:
```suggestion
Tips: It's recommended to increase the open file limit above 82920. WSL2 users may refer to [the comment]( https://github.com/Microsoft/WSL/issues/1688#issuecomment-532767317) if having difficulty in changing the `ulimit`.
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (jackwener) [inline CONTRIBUTING.md:110]:
```suggestion
Use [PD](https://github.com/tikv/pd) to manage the cluster (even if just one node on a single machine). 
```

**Comment by jackwener** [comment]
@sticnarf thanks for your review, I have fixed it.

**Review by sticnarf — APPROVED** [review]
LGTM.

@zhangyangyu PTAL

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -105,7 +105,13 @@ When building with make, cargo will automatically use [pipelined][p] compilation
 
 ## Running TiKV
 
-To run TiKV as an actual key-value store, you will need to run it as a cluster (a cluster can have just one node, which is useful for testing). You can do this on a single machine or on multiple machines. You need to use [PD](https://github.com/tikv/pd) to manage the cluster (even if there is just one node on a single machine). Instructions are in our [docs](https://tikv.org/docs/dev/tasks/deploy/binary/) (if you build TiKV from source, then you don't need to download the binary).
+To run TiKV as an actual key-value store, you will need to run it as a cluster (a cluster can have just one node, which is useful for testing). You can do this on a single machine or on multiple machines. 
+
+Use [PD](https://github.com/tikv/pd) to manage the cluster (even if just one node on a single machine). 
+
+Instructions are in our [docs](https://tikv.org/docs/dev/tasks/deploy/binary/) (if you build TiKV from source, you  could skip `1. Download package` and `tikv-server` is in directory `/target`).
```
Comment (zhangyangyu) [inline CONTRIBUTING.md:112]:
```suggestion
Instructions are in our [docs](https://tikv.org/docs/dev/tasks/deploy/binary/) (if you build TiKV from source, you could skip `1. Download package` and `tikv-server` is in directory `/target`).
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -105,7 +105,13 @@ When building with make, cargo will automatically use [pipelined][p] compilation
...
+
+Tips: It's recommended to increase the open file limit above 82920. WSL2 users may refer to [the comment]( https://github.com/Microsoft/WSL/issues/1688#issuecomment-532767317) if having difficulty in changing the `ulimit`.
```
Comment (zhangyangyu) [inline CONTRIBUTING.md:114]:
```suggestion
Tips: It's recommended to increase the open file limit above 82920. WSL2 users may refer to [the comment](https://github.com/Microsoft/WSL/issues/1688#issuecomment-532767317) if having difficulty in changing the `ulimit`.
```

**Review by zhangyangyu — APPROVED** [review]

**Comment by sticnarf** [comment]
@jackwener Each pull request is required to be linked to an issue. Can you create an enhancement issue and link it to this PR?

**Comment by jackwener** [comment]
😀It's ok.

**Review by 5kbpers — APPROVED** [review]

**Pull request merged**